packages/next/server/render.tsx

```diff
@@ -718,7 +721,11 @@ export async function renderToHTML(
             value={(moduleName) => reactLoadableModules.push(moduleName)}
           >
             <StyleRegistry registry={jsxStyleRegistry}>
-              {children}
+              <RuntimeImageConfigContext.Provider
+                value={images ? JSON.parse(images) : null}
```

Comment:
You shouldn't need to use JSON.parse(), the object should be passed as-is to the renderOpts

Reply:
Done, JSON.parse removed.

Reply:
I found the `renderOpts.images` is `string` during SSR, so I have to use `JSON.parse()`,

Am I miss something?

![ebab93c4bf75c488f57522f3c163380](https://user-images.githubusercontent.com/14243906/152645869-16f3f052-cbd5-4ef3-bb89-a40664079c6d.png)

Reply:
Thats very odd. The other properties on renderOpts support non-string types so there must be something missing. What about assigning that property `images: process.env.__NEXT_IMAGE_OPTS` everywhere the `const renderOpts = { ...` is used? That way webpack can replace the env var with the object.

Reply:
Also, what happens when you try `value={process.env.__NEXT_IMAGE_OPTS as any as ImageConfigComplete}` here like you did the for client?

Reply:
`renderOpts.image` was set to string in `base-server.ts` :
https://github.com/vercel/next.js/blob/938eb0e28c2615f64c1d26ea400b7a2cfb562dbe/packages/next/server/base-server.ts#L298

[When substitute`images: JSON.stringify(this.nextConfig.images),` with `images: this.nextConfig.images,` , there is no error, it passed all test case](https://github.com/JuniorTour/next.js/pull/4/files#diff-2b8e3503c1bebf856610a5551d9c5996435b99f45db389d4eeac1b671518875eL298)

<hr/>

So Can I remove the `JSON.stringify()` for `images: JSON.stringify(this.nextConfig.images),` in `base-server.ts` ?

So that we can use `<RuntimeImageConfigContext.Provider value={images} />` without JSON.parse().

Reply:
I've pushed code after remove the `JSON.stringify()` for `images: JSON.stringify(this.nextConfig.images),` in `base-server.ts`

If anything wrong, just comment.